### DISCUSSION TITLE
My GitHub repositories are not known by Google

### DISCUSSION THREAD

**Discussion by pgl10** [community/community#23571] [New to GitHub]
<p>Hello,<br>
I am the owner of <a href="https://github.com/pgl10" rel="nofollow noopener">https://github.com/pgl10</a> where I have two repositories. I have been trying many ways in order to Google being able to find any of them, but I never have gotten a good answer. At <a href="http://pgl10.chez.com/mathematiques.html" rel="nofollow noopener">http://pgl10.chez.com/mathematiques.html</a> which is well known by Google my two GitHub repositories have links to them.<br>
Please let me know what I can do to have them known by Google. Thank you very much. Best regards, pgl10</p>

**Comment by mpboom** [comment]
<p>Put up a <a href="https://pages.github.com/" rel="nofollow noopener">GitHub Pages</a> site!</p>

**Reply by SB0519** [reply]
Sorry buy I don't understand when go to that link it want you to create a repository but I already have 2 of them there should just be a setting to tick to enable google search, this is annoying.

**Reply by mpboom** [reply]
Well, Google will index your (public) GitHub repository eventually, but if you have specific needs surrounding information that is published and indexed by Google, you should put up a (GitHub Pages) website instead. There is no box to tick, because repositories that are public are indexable by Google by default.

**Comment by pgl10** [comment]
<p>Hello Mark,</p>
<p>Thank you very much for your answer.</p>
<p>I have done : <a href="https://pgl10.github.io" rel="nofollow noopener">https://pgl10.github.io</a></p>
<p>Is it what you have explained to me ?</p>
<p>Is it enough in order to have Google knowing my reposoteries ?</p>
<p>Best regards, pgl10</p>

**Comment by mpboom** [comment, marked as answer]
<p>Yeah, that should help a bit. Let me explain.</p>
<p>GitHub isn’t a place to promote repositories. GitHub won’t actively tell Google <strong>not</strong>  to list repositories, but there is a chance the only way to find your repo if you literally type in the name.</p>
<p>If you set up a GH pages site - which is an actual web site - with more content and information about the repo, chances are higher that people can also find your repos trough Google by searching with more general terms.</p>
<p>As an example: if people look for your “Opera” repo and search for <strong>big rational numbers library</strong> , chances are much higher they will find your repo if you have a GH pages site with more content and info related to the repo.</p>
<p>That said, your current setup with a simple listing will help a bit, but not much. Why not put some background info and documentation up there?</p>

**Reply by Antonomer** [reply]
How the .yml files help to appear your GH page in Google search ?

**Reply by mpboom** [reply]
I'm not sure what you're getting at @Antonomer, please elaborate.

**Reply by me9hanics** [reply]
I think he meant to reply to another comment (see mamutalib's comment).

**Comment by pgl10** [comment]
<p>Hello Mark,<br>
             <br>
Thank you very much again.<br>
Your answer will be very helpful for me and for other GitHub users.<br>
               <br>
Until now Google had never been able to find my GitHub repositories. But it is new, Google is finding them. So, may be, I will be able to improve the Google answers but they are already good as they are today.<br>
       <br>
Best regards.<br>
Au revoir, pgl10</p>

**Comment by mutalibcs** [comment]
<p>You should have setup a  _config.yml file for making the repository searchable.</p>

**Comment by shsaronian** [comment]
So every repository should have an individual _config.yml? I've seen people without a github pages or config file in their repos to get indexed. Like there are some github accounts that don't look as active as you'd expect them but are still searchable by Google. How is that possible?

**Reply by mutalibcs** [reply]
> So every repository should have an individual _config.yml? I've seen people without a github pages or config file in their repos to get indexed. Like there are some github accounts that don't look as active as you'd expect them but are still searchable by Google. How is that possible?

It's true. Sometimes, .yml file help to crawel the page in search engine.

**Comment by taoliujun** [comment]
so have you complete this problem?

I also have a repo and want google to know it's issue.

**Reply by me9hanics** [reply]
As others said, open a GitHub page, and you'll see it appear soon
Possibly you can add a _config.yml file to your repos.

**Reply by rh45-one** [reply]
Does the `_config.yml` file need to contain anything specific or does it simply need to exist?

**Comment by yesanqian** [comment]
In fact, about a month ago, I searched for relevant information on this topic.

I've created a GitHub repository for those who want to bypass Telegram channel or group copy-forward restrictions (e.g., restricted saving/downloading of content)(Restrict saving content,Telegram  download restricions):

https://github.com/yesanqian/tg-message-clone

As you can see, it has now been successfully indexed by Google.

**First, I want to sincerely thank everyone who offered advice on this matter!**

Here are the steps I took to help with the indexing:

> 1. I updated my profile and uploaded a new avatar.
> 
> 2. I shared the repository link on other Google-indexed sites (for better visibility and backlinks, which may take time to take effect). 
> 
> 3. I participated in some interactions on GitHub, such as joining discussions.
> 
> 4. I forked someone else's project.
> 
> 5. I waited for about a month.

**Hopefully, these steps will help others who are trying to solve the same problem.**

**Reply by IncrediDev** [reply]
Thanks
